jenkins/Jenkinsfile

```diff
@@ -1,374 +1,496 @@
+                imagesChanged.add(image)
+            }
+        }
     }
-  }
 
-  return !imagesChanged.isEmpty()
+    return !imagesChanged.isEmpty()
 }
 
-//if an image is in imageNames but not the confFile, the returned version will be an empty string
-def LinkedHashMap<String,String> getVersionMap(ArrayList<String> imageNames) {
-  def LinkedHashMap<String, String> tempVersionMap = [:]
-  def LinkedHashMap<String, String> finalVersionMap = [:]
-
-  def String[] configPaths = readFile("jenkins/confLocations.txt").trim().split("\n")
-  for (String configFile: configPaths) {
-
-    tempVersionMap = getVersionMap(imageNames, configFile)
-    for (String image in tempVersionMap.keySet()) {
-      if (finalVersionMap.keySet().contains(image) && !finalVersionMap[image].isEmpty()) {
-          //there is a nonempty version, we don't do anything here
-      } else {
-          finalVersionMap[image] = tempVersionMap[image]
-      }
+// If an image is in imageNames but not the confFile, the returned version will be an empty string
+LinkedHashMap<String,String> getVersionMap(Set<String> imageNames, String[] configPaths) {
+    LinkedHashMap<String, String> tempVersionMap = [:]
+    LinkedHashMap<String, String> finalVersionMap = [:]
+
+    for (String configFile: configPaths) {
+        tempVersionMap = getVersionMap(imageNames, configFile)
+        for (String image in tempVersionMap.keySet()) {
+            if (finalVersionMap.keySet().contains(image) && !finalVersionMap[image].isEmpty()) {
+                // there is a nonempty version, we don't do anything here
+            } else {
+                finalVersionMap[image] = tempVersionMap[image]
+            }
+        }
     }
-  }
-  return finalVersionMap
+    return finalVersionMap
 }
 
-def LinkedHashMap<String,String> getVersionMap(ArrayList<String> imageNames, String configFile) {
-  def LinkedHashMap<String,String> versionMap = [:]
+LinkedHashMap<String,String> getVersionMap(Set<String> imageNames, String configFile) {
+    LinkedHashMap<String,String> versionMap = [:]
 
-  //for example, if $image is 'welder-server', this line would isolate 'a4f0e04' from the line below in a $configFile
-  // welderDockerImage = "us.gcr.io/broad-dsp-gcr-public/welder-server:a4f0e04"
-  // the awk at the end ensures that the line can have comments or other whitespace after the terminating quote
-  for (String image: imageNames) {
-    def version = sh(
-      script: "cat $configFile | grep $image | tr ':' '\n' | tail -1 | tr -d '\"' | awk '{print \$1}' ",
-      returnStdout: true
-    ).trim()
+    // For example, if $image is 'welder-server', this line would isolate 'a4f0e04' from the line below in a $configFile.
+    // welderDockerImage = "us.gcr.io/broad-dsp-gcr-public/welder-server:a4f0e04"
+    // The awk at the end ensures that the line can have comments or other whitespace after the terminating quote.
+    // Escaping '$' inside the 'awk' command is to appease Jenkins.
```

Comment:
We must always aim to appease jenkins, for he is a fickle entity 🙈

Reply:
![image](https://user-images.githubusercontent.com/5368863/75935327-eb835680-5e4c-11ea-8d76-99a841bf8bf2.png)